doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
-an automatic buildbot runs the test suite after each code commit to confirm
-that yt hasn't broken recently.  To supplement this effort, we also maintain a
-`continuous integration server <https://tests.yt-project.org>`_ that runs the
-tests with each commit to the yt version control repository.
+checking in any code that breaks existing functionality. Additionally, as a safeguard, the test suite is automatically run after each commit that is pushed to the yt repository on github.
 
 .. _unit_testing:
 
 Unit Testing
 ------------
 
-What do Unit Tests Do
+What Do Unit Tests Do
 ^^^^^^^^^^^^^^^^^^^^^
 
-Unit tests are tests that operate on some small set of machinery, and verify
-that the machinery works.  yt uses the `Nose
-<https://nose.readthedocs.io/en/latest/>`_ framework for running unit tests.  In
-practice, what this means is that we write scripts that assert statements, and
-Nose identifies those scripts, runs them, and verifies that the assertions are
-true and the code runs without crashing.
+Unit tests are tests that operate on some small piece of machinery and verify
+that the machinery works. In
+practice, this means that we make assertions about a piece of machinery and then pytest runs the machinery, verifies that the assertions are true, and ensures that the code runs without crashing. An example of a unit test is :func:`~yt.fields.tests.test_fields.test_all_fields`.
 
 How to Run the Unit Tests
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-One can run the unit tests very straightforwardly from any python interpreter
-that can import the yt module:
-
-.. code-block:: python
-
-   import yt
-
-   yt.run_nose()
-
-If you are developing new functionality, it is sometimes more convenient to use
-the Nose command line interface, ``nosetests``. You can run the unit tests
-using ``nose`` by navigating to the base directory of the yt git
-repository and invoking ``nosetests``:
+One can run the unit tests by navigating to the base directory of the yt git
+repository and invoking ``pytest``:
 
 .. code-block:: bash
 
    $ cd $YT_GIT
-   $ nosetests
+   $ pytest
 
 where ``$YT_GIT`` is the path to the root of the yt git repository.
 
-If you want to specify a specific unit test to run (and not run the entire
-suite), you can do so by specifying the path of the test relative to the
-``$YT_GIT/yt`` directory -- note that you strip off one yt more than you
-normally would!  For example, if you want to run the plot_window tests, you'd
+If you only want to run tests in a specific file, you can do so by specifying the path of the test relative to the
+``$YT_GIT/`` directory. For example, if you want to run the ``plot_window`` tests, you'd
 run:
 
 .. code-block:: bash
 
-   $ nosetests yt/visualization/tests/test_plotwindow.py
+   $ pytest yt/visualization/tests/test_plotwindow.py
 
-Handling yt dependencies
-^^^^^^^^^^^^^^^^^^^^^^^^
+Additionally, if you only want to run, say, ``test_all_fields`` in ``plot_window.py``, you can do so by running:
```

Comment:
```suggestion
from the yt source code root directory. 
Additionally, if you only want to run a specific test in a
test file (rather than all of the tests contained in the file),
such as, ``test_all_fields`` in ``plot_window.py``, you can do so by running:
```